java/memory/memory-core/src/test/java/org/apache/arrow/memory/DefaultAllocationManagerFactory.java

```diff
@@ -0,0 +1,64 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.arrow.memory;
+
+import org.apache.arrow.memory.util.MemoryUtil;
+
+/**
+ * The default Allocation Manager Factory for a module.
+ *
+ * This will be split out to the arrow-memory-netty/arrow-memory-unsafe modules
+ * as part of ARROW-8230. This is currently a placeholder which defaults to Netty.
+ *
+ */
+public class DefaultAllocationManagerFactory implements AllocationManager.Factory {
```

Comment:
We have 3 classes in the code base named "DefaultAllocationManagerFactory". Do we need all of them?

Reply:
Hey @liyafan82, yes i think we do:

1) lives in the netty module.
2) lives in the unsafe module
3) lives in the tests portion of the core memory module.

On startup we first check if a specific memory allocator has been requested. If yes we instantiate that. If no specific impl has been requested then we search the classpath for a `DefaultAllocationManagerFactory`. If we find one then we instantiate it, else we throw a Runtime exception (can't continue without an allocation manager).

So each concrete impl of `AllocationManager` requires a factory to create it and we have a trivial allocation manager in `arrow-memory-core` tests to help the core tests finish.